src/sentry/taskworker/task.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from collections.abc import Callable
+from datetime import timedelta
+from functools import update_wrapper
+from typing import TYPE_CHECKING, Any
+
+from django.utils import timezone
+
+if TYPE_CHECKING:
+    from sentry_protos.sentry.v1.taskworker_pb2 import RetryState
+
+    from sentry.taskworker.registry import TaskNamespace
+    from sentry.taskworker.retry import Retry
```

Comment:
if this isn't an import cycle these should just be at the module scope -- if it is an import cycle perhaps these should be reorganized to *not* have a cycle

Reply:
There is one import cycle remaining because of types. `Task` has a reference `TaskNamespace` and the namespaces reference `Task`.